assets/entrypoint.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash -e
+# =====================================================================
+# Build script running OpenNMS in Docker environment
+#
+# Source: https://github.com/opennms-forge/docker-horizon-core-web
+# Web: https://www.opennms.org
+#
+# =====================================================================
+
+# Cause false/positives
+# shellcheck disable=SC2086
+
+OPENNMS_HOME="/opt/opennms"
+OPENNMS_OVERLAY="/opt/opennms-overlay"
+
+# Error codes
+E_ILLEGAL_ARGS=126
+E_INIT_CONFIG=127
+E_DEPRECATED_CONFIG=128
+
+# Help function used in error messages and -h option
+usage() {
+  echo ""
+  echo "Docker entry script for OpenNMS service container"
+  echo ""
+  echo "Overlay Config file:"
+  echo "If you want to overwrite the default configuration with your custom config, you can use an overlay"
+  echo "folder which needs to be mounted to ${OPENNMS_OVERLAY}."
+  echo "Every file in this folder is overwriting files in ${OPENNMS_HOME}."
+  echo ""
+  echo "-h: Show this help."
+  echo "-i: Initialize or update the database and OpenNMS configuration files, runs install -dis."
+  echo "    OpenNMS will be *NOT* started"
+  echo "-s: Start OpenNMS in foreground with an existing configuration."
+  echo "-t: options: Run the config-tester, default is -h to show usage."
+  echo ""
+}
+
+install() {
+  echo "Run OpenNMS install command to initialize or upgrade the database schema and configurations."
+  ${JAVA_HOME}/bin/java -Dopennms.home="${OPENNMS_HOME}" -Dlog4j.configurationFile="${OPENNMS_HOME}"/etc/log4j2-tools.xml -cp "${OPENNMS_HOME}/lib/opennms_bootstrap.jar" org.opennms.bootstrap.InstallerBootstrap "${@}" || exit ${E_INIT_CONFIG}
+}
+
+initNewts() {
+  if [ -n "${INIT_NEWTS}" ]; then
+    REPLICATION_FACTOR=${REPLICATION_FACTOR-1}
+    ${JAVA_HOME}/bin/java -Dopennms.manager.class="org.opennms.netmgt.newts.cli.Newts" -Dopennms.home="${OPENNMS_HOME}" -Dlog4j.configurationFile="${OPENNMS_HOME}"/etc/log4j2-tools.xml -jar ${OPENNMS_HOME}/lib/opennms_bootstrap.jar init -r ${REPLICATION_FACTOR} || exit ${E_INIT_CONFIG}
+  fi
+}
+
+configTester() {
+  echo "Run config tester to validate existing configuration files."
+  ${JAVA_HOME}/bin/java -Dopennms.manager.class="org.opennms.netmgt.config.tester.ConfigTester" -Dopennms.home="${OPENNMS_HOME}" -Dlog4j.configurationFile="${OPENNMS_HOME}"/etc/log4j2-tools.xml -jar ${OPENNMS_HOME}/lib/opennms_bootstrap.jar "${@}" || exit ${E_INIT_CONFIG}
+}
+
+processConfdTemplates() {
+  echo "Processing confd templates using /etc/confd/confd.toml"
```

Comment:
I guess in the future it might be nice to be able to overlay a confd toml, to point to an external keystore for example. However for now I don't think that is an issue.